### PR TITLE
Nested resigning

### PR DESCRIPTION
**Pull request by baarde**
Sigh now supports passing multiple provisioning profiles to the resign action (fastlane/sigh#174).

### CODE REVIEW

File: lib/fastlane/actions/resign.rb [hunk 1]
```diff
@@ -33,8 +47,12 @@ def self.available_options
                                        env_name: "FL_RESIGN_PROVISIONING_PROFILE",
                                        description: "Path to your provisioning_profile. Optional if you use `sigh`",
                                        default_value: Actions.lane_context[SharedValues::SIGH_PROFILE_PATH],
+                                       is_string: false,
                                        verify_block: proc do |value|
-                                         raise "No provisioning_profile file given or found, pass using `provisioning_profile: 'path/app.mobileprovision'`".red unless File.exist?(value)
+                                         files = value.kind_of?(Enumerable) ? value.map { |fst, snd| snd || fst } : [value]
```
Comment (KrauseFx) [inline lib/fastlane/actions/resign.rb]: That line is confusing, can we make it simpler?

**Comment by KrauseFx** [comment]
Looks good, I added a comment before it's ready to be merged :+1:

File: lib/fastlane/actions/resign.rb (hunk 1 above)
Comment (baarde) [inline lib/fastlane/actions/resign.rb]: We could write something like this:
```ruby
files = case value
when Enumerable; value.map { |a, b| b || a }
else [value]
end
```
I know the `map` is a little bit confusing. But It's the simplest to support both hashes and arrays of strings and also arrays of strings and tuples. For example, we could imagine passing the following value:
```ruby
["app.mobileprovision", ["com.example.app.extension", "ext.mobileprovision"]]
```
That would be translated as:
```
-p app.mobileprovision -p com.example.app.extension=ext.mobileprovision
```
Or we can assume that nobody needs that. In which case, I propose this:
```ruby
files = case value
when Hash; value.values
when Enumerable; value
else [value]
end
```

Comment (KrauseFx) [inline lib/fastlane/actions/resign.rb]: Let's keep it simple and have one way to pass the parameters.

File: lib/fastlane/actions/resign.rb [hunk 2]
```diff
@@ -17,6 +17,20 @@ def self.description
...
       end
 
+      def self.details
+        [
+          "You may provide multiple provisioning profiles if the application contains",
+          "nested applications or app extensions, which need their own provisioning",
+          "profile. You can do so by passing an array of provisiong profile strings or a",
+          "hash that associates provisioning profile values to bundle identifier keys.",
+          "",
+          "resign ipa: 'path', signing_identity: 'identity', provisioning_profile: {",
```
Comment (KrauseFx) [inline lib/fastlane/actions/resign.rb]: Can you change that to use the hash syntax? 
```ruby
resign(ipa: "path", signing...)
```

File: lib/fastlane/actions/resign.rb [hunk 3]
```diff
@@ -17,6 +17,20 @@ def self.description
...
+          "  'com.example.awesome-app' => 'App.mobileprovision',",
+          "  'com.example.awesome-app.app-extension' => 'Extension.mobileprovision'",
```
Comment (KrauseFx) [inline lib/fastlane/actions/resign.rb]: Can you also update `Actions.md` to describe this?

**Comment by rompic** [comment]
+1

**Comment by KrauseFx** [comment]
Looks great, thanks for updating the PR :+1:

**Pull request merged**